### PR TITLE
[android] Remove Android Studio warnings

### PR DESCRIPTION
**Pull request by Jean-BaptisteC**
This PR fixes some Android Studio warnings:
- Remove unused imports
- Replace StringBuilder with String
- Use lambda
- Replace `.size == 0` by `isEmpty()`
- Remove useless semicolons
- Remove redundant type
- Use final on variables when possible

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/bookmarks/Holders.java [hunk 1]
```diff
@@ -406,11 +406,10 @@ void bind(@NonNull SectionPosition position,
       final long trackId = sectionsDataSource.getTrackId(position);
       Track track = BookmarkManager.INSTANCE.getTrack(trackId);
       mName.setText(track.getName());
-      mDistance.setText(new StringBuilder().append(mDistance.getContext()
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/bookmarks/Holders.java:409]: 1. Was it done deliberately to speed up the code? Is there any comment in git blame?
2. Does it make sense to cache .getString(R.string.length) result here and maybe in some other places, where it is called repeatedly for many objects?

File: android/app/src/main/java/app/organicmaps/bookmarks/data/BookmarkCategory.java [hunk 2]
```diff
@@ -105,17 +105,15 @@ public String getDescription()
   @Override
   public String toString()
   {
-    final StringBuilder sb = new StringBuilder("BookmarkCategory{");
-    sb.append("mId=").append(mId);
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/bookmarks/data/BookmarkCategory.java:109]: Are you sure that the nee code is faster? Is there any documentation or benchmark about it?

File: android/app/src/main/java/app/organicmaps/editor/FeatureCategoryFragment.java [hunk 3]
```diff
@@ -102,8 +103,7 @@ private FeatureCategory[] makeFeatureCategoriesFromTypes(@NonNull String[] creat
       categories[i] = new FeatureCategory(creatableTypes[i], localizedType);
     }
 
-    Arrays.sort(categories, (lhs, rhs) ->
-      lhs.getLocalizedTypeName().compareTo(rhs.getLocalizedTypeName()));
+    Arrays.sort(categories, Comparator.comparing(FeatureCategory::getLocalizedTypeName));
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/editor/FeatureCategoryFragment.java:106]: Curious, why is it better?

File: android/app/src/main/java/app/organicmaps/util/log/LogsManager.java [hunk 4]
```diff
@@ -278,29 +278,26 @@ public static String getMemoryInfo(@NonNull Context context)
     final ActivityManager activityManager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
     activityManager.getMemoryInfo(mi);
 
-    final StringBuilder log = new StringBuilder(256);
-    log.append("Memory info: ")
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/util/log/LogsManager.java:282]: Ditto

File: android/app/src/main/java/app/organicmaps/bookmarks/Holders.java (hunk 1 above)
Comment (Jean-BaptisteC) [reply android/app/src/main/java/app/organicmaps/bookmarks/Holders.java]: Nothing in git blame
It's the only place using `.getString(R.string.length)`

File: android/app/src/main/java/app/organicmaps/bookmarks/data/BookmarkCategory.java (hunk 2 above)
Comment (Jean-BaptisteC) [reply android/app/src/main/java/app/organicmaps/bookmarks/data/BookmarkCategory.java]: You're right
In the most of case, StringBuilder is more efficient -> https://dip-mazumder.medium.com/stringbuilder-vs-string-in-java-a-guide-for-optimal-memory-usage-4a284d8243ea

File: android/app/src/main/java/app/organicmaps/MwmActivity.java [hunk 5]
```diff
@@ -318,9 +317,7 @@ private void migrateOAuthCredentials()
 
       // Notify user to re-login
       dismissAlertDialog();
-      final DialogInterface.OnClickListener navigateToLoginHandler = (DialogInterface dialog, int which) -> {
-        startActivity(new Intent(MwmActivity.this, OsmLoginActivity.class));
-      };
+      final DialogInterface.OnClickListener navigateToLoginHandler = (DialogInterface dialog, int which) -> startActivity(new Intent(MwmActivity.this, OsmLoginActivity.class));
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/MwmActivity.java:320]:
```java
final DialogInterface.OnClickListener navigateToLoginHandler = (dialog, which) -> startActivity(new Intent(MwmActivity.this, OsmLoginActivity.class));
```

or

```java
final DialogInterface.OnClickListener navigateToLoginHandler = (_, _) -> startActivity(new Intent(MwmActivity.this, OsmLoginActivity.class));
```

**Pull request merged**